BEIMA.Backend.FT/BeimaClient.cs

```diff
@@ -118,6 +119,55 @@ public async Task<HttpResponseMessage> SendRequest(string route, HttpVerb verb,
             return response;
         }
 
+        /// <summary>
+        /// Sends an multipart form data http request to the backend server and returns the http response.
+        /// </summary>
+        /// <param name="route">The url route excluding the base address.</param>
+        /// <param name="data">The object to send with the request.</param>
+        /// <param name="files">List of files to send with the request</param>
+        /// <returns>The http response message of the request.</returns>
+        /// <exception cref="HttpRequestException"></exception>
+        public async Task<HttpResponseMessage> SendMultiPartRequest(string route, Object data, FormFileCollection? files = null)
+        {
+            StringContent json;
+            HttpResponseMessage response;
+
+            if (data is string)
+            {
+                json = new StringContent((string)data);
+            }
+            else
+            {
+                var jsonObject = JsonConvert.SerializeObject(data);
+                json = new StringContent(jsonObject, Encoding.UTF8, "application/json");
+            }
+
+            MultipartFormDataContent form = new MultipartFormDataContent();
+
+            form.Add(json, "data");
+            if (files != null)
+            {
+                foreach (var file in files)
+                {
+                    form.Add(new StreamContent(file.OpenReadStream()), file.Name, file.FileName);
```

Comment:
So if two files have the same file.Name, (i.e. we have two documents) will they be added to the same file bucket? Or will one replace the other.

Reply:
They will be added to the same file bucket, but they will be stored based off of the uuid generated by the storage provider. Every file object in a device's file list will store their original filename as well as the uuid we generate. That way we can allow non-unique file names to be used. File.Name is different from File.FileName though. File.Name is sort of a tag used on the multipart form request and isn't really used elsewhere.